builtin/providers/google/provider.go

```diff
@@ -250,3 +251,15 @@ func getNetworkNameFromSelfLink(network string) (string, error) {
 
 	return network, nil
 }
+
+func readError(err error, d *schema.ResourceData, resource string) error {
```

Comment:
nit: maybe `handleNotFoundError`? `readError` doesn't really indicate this is for 404s, to me.